creator-node/src/snapbackSM.js

```diff
@@ -47,63 +61,387 @@ class SnapbackSM {
+            // Enqueue sync for secondary1 if required
+            if (secondary1SyncRequired && secondary1 != null) {
+              await this.issueSecondarySync(userWallet, secondary1, this.endpoint)
+              numSyncsIssued += 1
+            }
+            // Enqueue sync for secondary2 if required
+            if (secondary2SyncRequired && secondary2 != null) {
+              await this.issueSecondarySync(userWallet, secondary2, this.endpoint)
+              numSyncsIssued += 1
+            }
+          } catch (e) {
+            this.log(`Caught error for user ${user.wallet}, ${JSON.stringify(user)}, ${e.message}`)
+          }
+        }
+      )
+    )
+
+    let previousModuloSlice = this.currentModuloSlice
+
+    // Increment and adjust current slice by ModuloBase
+    this.currentModuloSlice += 1
+    this.currentModuloSlice = this.currentModuloSlice % ModuloBase
+
+    this.log(`Updated modulo slice from ${previousModuloSlice} to ${this.currentModuloSlice}`)
+    this.log(`Issued ${numSyncsIssued} sync ops`)
+    this.log(`------------------END Process SnapbackSM Operation, slice ${previousModuloSlice} ------------------`)
+  }
+
+  // Track an ongoing sync operation to a secondary
+  async monitorSecondarySync (syncWallet, primaryClockValue, secondaryUrl) {
+    let startTime = Date.now()
+    // Monitor the sync status
+    let syncMonitoringRequestParameters = {
+      method: 'get',
+      baseURL: secondaryUrl,
+      url: `/sync_status/${syncWallet}`,
+      responseType: 'json'
+    }
+    let syncAttemptCompleted = false
+    while (!syncAttemptCompleted) {
       try {
-        await this.processStateMachineOperation(job)
+        let syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+        let respData = syncMonitoringResp.data.data
+        this.log(`processSync ${syncWallet} secondary response: ${JSON.stringify(respData)}`)
+        // A success response does not necessarily guarantee completion, validate response data to confirm
+        if (respData.clockValue === primaryClockValue) {
+          syncAttemptCompleted = true
+          this.log(`processSync ${syncWallet} clockValue from secondary:${respData.clockValue}, primary:${primaryClockValue}`)
+        }
       } catch (e) {
-        logger.info(`Error processing ${e}`)
-      } finally {
-        this.log(`DEV MODE next job in ${DevDelayInMS}ms at ${new Date(Date.now() + DevDelayInMS)}`)
-        await utils.timeout(DevDelayInMS)
-        this.stateMachineQueue.add({ startTime: Date.now() })
-        done()
+        this.log(`processSync ${syncWallet} error querying sync_status: ${e}`)
+      }
+      if (Date.now() - startTime > MaxSyncMonitoringDurationInMs) {
+        this.log(`ERROR: processSync ${syncWallet} timeout for ${syncWallet}`)
+        syncAttemptCompleted = true
       }
+      // 1s delay between retries
+      await utils.timeout(1000)
```

Comment:
i do think this is too aggressive. we should maybe make this 15 seconds? considering this is a background process and we probably don't want to saturate the secondary db's with up to 10 additional db calls a second

Reply:
agreed, i think that this is a reasonable step